src/index.js

```diff
@@ -153,23 +175,36 @@ export default class Firedux {
     return new Promise((resolve, reject) => {
       dispatch({type: 'FIREBASE_LOGIN_ATTEMPT'})
 
+      const handleError = function (error, authData = {}) {
+        console.error('FB AUTH ERROR', error, authData)
+        dispatch({type: 'FIREBASE_LOGIN_ERROR', error})
+        reject(error)
+      }
+
       const handler = function (error, authData) {
         // TODO: Error handling.
         debug('FB AUTH', error, authData)
-        if (error) {
-          console.error('FB AUTH ERROR', error, authData)
-          dispatch({type: 'FIREBASE_LOGIN_ERROR', error})
-          reject(error)
-          return
-        }
-        localStorage.setItem('FIREBASE_TOKEN', authData.token)
+
+        if (error) return handleError(error)
+
+        localStorage.setItem('FIREBASE_TOKEN', (authData.token || authData.refreshToken))
         that.authData = authData
         dispatch({type: 'FIREBASE_LOGIN', authData: authData, error: error})
         resolve(authData)
       }
 
       try {
-        if (credentials.token) {
+        if (this.v3) {
+          // need to ignore redux
+          if (_.isFunction(credentials)) return null
+
+          // TODO add custom later...
+          this.auth()
+            .signInWithEmailAndPassword(credentials.email, credentials.password)
+            .then(authData => handler(null, authData))
+            .catch(error => handleError(error))
+        }
+        else if (credentials.token) { // eslint-disable-line brace-style
```

Comment:
Don't violate eslint standard here - collapse with preceding brace to single line.